components/selection/selection-action-mixin.js

```diff
@@ -24,7 +33,16 @@ export const SelectionActionMixin = superclass => class extends SelectionObserve
 
 	set selectionInfo(value) {
 		super.selectionInfo = value;
-		this.disabled = (this.requiresSelection && this.selectionInfo.state === SelectionInfo.states.none);
+		if (this.selectionInfo.keys.length > this.maxSelectionCount) {
```

Comment:
This does not handle the "select all pages" case. I am going to try a change to address the edge case where that is enabled, the user has selected it, but the page size is less than the `max-selection-count`.

Reply:
Managed to add check for this case. If the consumer doesn't specify `itemCount` on pageable then there's not much more we can do, but I think that is required for select-all-pages to be enabled anyway.